test/wallet-auction-test.js

```diff
@@ -0,0 +1,191 @@
+  enabled: false
+});
+
+const chain = new Chain({
+  memory: true,
+  network,
+  workers
+});
+
+const miner = new Miner({
+  chain,
+  workers
+});
+
+const cpu = miner.cpu;
+
+const wdb = new WalletDB({
+  network: network,
+  workers: workers
+});
+
+describe('Wallet Auction', function() {
+  this.timeout(15000);
+
+  let winner;
+  const currentCBMaturity = network.coinbaseMaturity;
+
+  before(() => {
+    network.coinbaseMaturity = 1;
+  });
+
+  after(() => {
+    network.coinbaseMaturity = currentCBMaturity;
+  });
+
+  it('should open chain, miner and wallet', async () => {
+    await chain.open();
+    await miner.open();
+    await wdb.open();
+
+    winner = await wdb.create();
+
+    chain.on('connect', async (entry, block) => {
+      await wdb.addBlock(entry, block.txs);
+    });
+  });
+
+  it('should add addrs to miner', async () => {
+    const addr = await winner.createReceive();
+    miner.addresses = [addr.getAddress().toString(network)];
+  });
+
+  it('should mine 20 blocks', async () => {
+    for (let i = 0; i < 20; i++) {
+      const block = await cpu.mineBlock();
+      assert(block);
+      assert(await chain.add(block));
+    }
+  });
+
+  let openAuctionMTX;
+  it('should open auction', async () => {
+    openAuctionMTX = await winner.createOpen(NAME1, false);
+    await winner.sign(openAuctionMTX);
+    const tx = openAuctionMTX.toTX();
+    await wdb.addTX(tx);
+  });
+
+  it('should fail to create duplicate open', async () => {
+    let err;
+    try {
+      await winner.createOpen(NAME1, false);
+    } catch (e) {
+      err = e;
+    }
+
+    assert(err);
+    assert.strictEqual(err.message, `Already sent an open for: ${NAME1}.`);
+  });
+
+  it('should mine 1 block', async () => {
+    const job = await cpu.createJob();
+    job.addTX(openAuctionMTX.toTX(), openAuctionMTX.view);
+    job.refresh();
+
+    const block = await job.mineAsync();
+
+    assert(await chain.add(block));
+  });
+
+  it('should fail to re-open auction during OPEN phase', async () => {
+    let err;
+    try {
+      await winner.createOpen(NAME1, false);
+    } catch (e) {
+      err = e;
+    }
+
+    assert(err);
+    assert.strictEqual(err.message, 'Name is already opening.');
+  });
+
+  it('should mine enough blocks to enter BIDDING phase', async () => {
+    for (
+      let i = 0;
+      i < network.names.treeInterval;
+      i++
+    ) {
+      const block = await cpu.mineBlock();
+      assert(block);
+      assert(await chain.add(block));
+    }
+  });
+
+  it('should fail to re-open auction during BIDDING phase', async () => {
+    let err;
+    try {
+      await winner.createOpen(NAME1, false);
+    } catch (e) {
+      err = e;
+    }
+
+    assert(err);
+    assert.strictEqual(err.message, 'Name is not available.');
+  });
+
+  it('should mine enough blocks to expire auction', async () => {
+    for (
+      let i = 0;
+      i < network.names.biddingPeriod + network.names.revealPeriod;
```

Comment:
Nit: precalculate `network.names.biddingPeriod + network.names.revealPeriod` into a variable as to not need to use multiple lines for the `for` statement